core/src/apps/cardano/sign_tx.py

```diff
@@ -96,44 +99,52 @@ async def sign_tx(
 
     validate_network_info(msg.network_id, msg.protocol_magic)
 
-    if _has_stake_pool_registration(msg):
-        return await _sign_stake_pool_registration_tx(ctx, msg, keychain)
-    else:
-        return await _sign_standard_tx(ctx, msg, keychain)
-
-
-async def _sign_standard_tx(
-    ctx: wire.Context, msg: CardanoSignTx, keychain: seed.Keychain
-) -> CardanoSignedTx:
     try:
-        for i in msg.inputs:
-            await validate_path(
-                ctx, keychain, i.address_n, SCHEMA_ADDRESS.match(i.address_n)
+        if _has_stake_pool_registration(msg):
+            cborized_tx, tx_hash = await _sign_stake_pool_registration_tx(
+                ctx, msg, keychain
             )
+        else:
+            cborized_tx, tx_hash = await _sign_ordinary_tx(ctx, msg, keychain)
 
-        _validate_outputs(keychain, msg.outputs, msg.protocol_magic, msg.network_id)
-        _validate_certificates(msg.certificates, msg.protocol_magic, msg.network_id)
-        _validate_withdrawals(msg.withdrawals)
-        _validate_metadata(msg.metadata)
+        signed_tx_chunks = _get_signed_tx_chunks(cborized_tx, tx_hash)
 
-        # display the transaction in UI
-        await _show_standard_tx(ctx, keychain, msg)
+        for tx_chunk_message in signed_tx_chunks:
+            if tx_chunk_message.expect_more_chunks:
```

Comment:
I am not sure we want to set it at this level. It would make more sense to construct the message here based on chunks

Reply:
do you have in mind constructing the protobuf message right here and setting the "expect_more_chunks" flag based on a StopIteration error from the underlying chunk generator? I agree that would be cleaner indeed, it would also remove the need to pass the tx_hash to the function that produces the chunks